asserts/snap_asserts.go

```diff
@@ -316,3 +316,131 @@ func assembleSnapRevision(assert assertionBase) (Assertion, error) {
 		timestamp:     timestamp,
 	}, nil
 }
+
+// Validation holds a validation assertion, describing that a combination of
+// (snap-id, approved-snap-id, approved-revision) has been validated for
```

Comment:
What is the difference between the `snap-id` and `approved-snap-id` in this context?